### ISSUE TITLE
2.15 & 2.16 v3 (Windows): eth+ethb3 ethash never starts mining (RX5700)

### ISSUE THREAD

**Issue by UselessGuru** [sp-hash/TeamBlackMiner#432]
Same as https://github.com/sp-hash/TeamBlackMiner/issues/428

Issue is still not fixed:

```
00:00:00 [2023-12-15 15:01:09.262]
00:00:00 [2023-12-15 15:01:10.045] OpenCL driver version: 546.1
00:00:00 [2023-12-15 15:01:10.094] Cuda driver version: 12.3
00:00:00 [2023-12-15 15:01:10.094] Cuda runtime API version: 12.0
00:00:00 [2023-12-15 15:01:10.094]
00:00:00 [2023-12-15 15:01:10.094]
00:00:00 [2023-12-15 15:01:10.094]
00:00:00 [2023-12-15 15:01:10.094] Welcome to Team Black Miner 2.15
00:00:00 [2023-12-15 15:01:10.094]
00:00:00 [2023-12-15 15:01:10.094]
00:00:00 [2023-12-15 15:01:10.094] GPU0 gfx1010:xnack- 8GB added to miner as OpenCL device
00:00:00 [2023-12-15 15:01:10.099] Trying ethashb3.eu.mine.zergpool.com:9996 (ethashb3)
00:00:00 [2023-12-15 15:01:10.100] Trying europe.ethash-hub.miningpoolhub.com:20605 (ethash)
00:00:00 [2023-12-15 15:01:10.100] Api server started at 127.0.0.1:4001
00:00:00 [2023-12-15 15:01:10.100] GPU0 Started CL work thread
00:00:00 [2023-12-15 15:01:10.101] GPU0 Started CL work thread
00:00:00 [2023-12-15 15:01:10.192] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to subscribe
00:00:00 [2023-12-15 15:01:10.223] europe.ethash-hub.miningpoolhub.com: Difficulty is set to 3.000000
00:00:00 [2023-12-15 15:01:10.254] GPU0 Starting generating ethash dag (Epoch: 213)
00:00:00 [2023-12-15 15:01:10.317] ethashb3.eu.mine.zergpool.com: Pool responded successfully to subscribe
00:00:00 [2023-12-15 15:01:10.410] ethashb3.eu.mine.zergpool.com: Pool responded successfully to extranonce subscribe
00:00:00 [2023-12-15 15:01:10.426] ethashb3.eu.mine.zergpool.com: Extranonce is set to 6508000000000000
00:00:00 [2023-12-15 15:01:10.490] ethashb3.eu.mine.zergpool.com: Pool responded successfully to authorize
00:00:00 [2023-12-15 15:01:10.505] ethashb3.eu.mine.zergpool.com: Difficulty is set to 0.249996
00:00:00 [2023-12-15 15:01:10.535] GPU0 Starting generating ethashb3 dag (Epoch: 45)
00:00:00 [2023-12-15 15:01:10.599] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to authorize
00:00:01 [2023-12-15 15:02:10.625] miningpoolhub.com (ethash)    PING: 44ms    DIFFICULTY: 3.00     EPOCH: 213
00:00:01 [2023-12-15 15:02:10.625]
00:00:01 [2023-12-15 15:02:10.625] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:01 [2023-12-15 15:02:10.625] GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
00:00:01 [2023-12-15 15:02:10.625]                                                                          0
00:00:01 [2023-12-15 15:02:10.625] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:01 [2023-12-15 15:02:10.625] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-12-15 15:02:10.625]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-12-15 15:02:10.625]
00:00:01 [2023-12-15 15:02:10.625] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:01 [2023-12-15 15:02:10.625]
00:00:01 [2023-12-15 15:02:10.625] eu.mine.zergpool.com (ethashb3)    PING: 94ms    DIFFICULTY: 0.250    EPOCH: 45
00:00:01 [2023-12-15 15:02:10.625]
00:00:01 [2023-12-15 15:02:10.625] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:01 [2023-12-15 15:02:10.625] GPU0    5700XT   OpenCL  -1      0       0/0     46      2040    1840    0
00:00:01 [2023-12-15 15:02:10.625]                                                                          0
00:00:01 [2023-12-15 15:02:10.625] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:01 [2023-12-15 15:02:10.625] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-12-15 15:02:10.625]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-12-15 15:02:10.625]
00:00:01 [2023-12-15 15:02:10.625] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:01 [2023-12-15 15:02:10.625]
00:00:02 [2023-12-15 15:03:10.629] miningpoolhub.com (ethash)    PING: 47ms    DIFFICULTY: 3.00     EPOCH: 213
00:00:02 [2023-12-15 15:03:10.629]
00:00:02 [2023-12-15 15:03:10.629] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:02 [2023-12-15 15:03:10.629] GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
00:00:02 [2023-12-15 15:03:10.629]                                                                          0
00:00:02 [2023-12-15 15:03:10.629] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:02 [2023-12-15 15:03:10.629] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:02 [2023-12-15 15:03:10.629]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:02 [2023-12-15 15:03:10.629]
00:00:02 [2023-12-15 15:03:10.629] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:02 [2023-12-15 15:03:10.629]
00:00:02 [2023-12-15 15:03:10.629] eu.mine.zergpool.com (ethashb3)    PING: 95ms    DIFFICULTY: 0.250    EPOCH: 45
00:00:02 [2023-12-15 15:03:10.629]
00:00:02 [2023-12-15 15:03:10.629] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:02 [2023-12-15 15:03:10.629] GPU0    5700XT   OpenCL  -1      0       0/0     46      2040    1840    0
00:00:02 [2023-12-15 15:03:10.629]                                                                          0
00:00:02 [2023-12-15 15:03:10.629] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:02 [2023-12-15 15:03:10.629] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:02 [2023-12-15 15:03:10.629]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:02 [2023-12-15 15:03:10.629]
00:00:02 [2023-12-15 15:03:10.629] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:02 [2023-12-15 15:03:10.629]
00:00:03 [2023-12-15 15:04:10.662] miningpoolhub.com (ethash)    PING: 48ms    DIFFICULTY: 3.00     EPOCH: 213
00:00:03 [2023-12-15 15:04:10.662]
00:00:03 [2023-12-15 15:04:10.662] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:03 [2023-12-15 15:04:10.662] GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
00:00:03 [2023-12-15 15:04:10.662]                                                                          0
00:00:03 [2023-12-15 15:04:10.662] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:03 [2023-12-15 15:04:10.662] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:03 [2023-12-15 15:04:10.662]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:03 [2023-12-15 15:04:10.662]
00:00:03 [2023-12-15 15:04:10.662] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:03 [2023-12-15 15:04:10.662]
00:00:03 [2023-12-15 15:04:10.662] eu.mine.zergpool.com (ethashb3)    PING: 91ms    DIFFICULTY: 0.250    EPOCH: 45
00:00:03 [2023-12-15 15:04:10.662]
00:00:03 [2023-12-15 15:04:10.662] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:03 [2023-12-15 15:04:10.662] GPU0    5700XT   OpenCL  -1      0       0/0     46      2040    1840    0
00:00:03 [2023-12-15 15:04:10.662]                                                                          0
00:00:03 [2023-12-15 15:04:10.662] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:03 [2023-12-15 15:04:10.662] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:03 [2023-12-15 15:04:10.662]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:03 [2023-12-15 15:04:10.662]
00:00:03 [2023-12-15 15:04:10.662] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:03 [2023-12-15 15:04:10.662]
00:00:04 [2023-12-15 15:05:10.622] miningpoolhub.com (ethash)    PING: 49ms    DIFFICULTY: 3.00     EPOCH: 213
00:00:04 [2023-12-15 15:05:10.622]
00:00:04 [2023-12-15 15:05:10.622] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:04 [2023-12-15 15:05:10.622] GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
00:00:04 [2023-12-15 15:05:10.622]                                                                          0
00:00:04 [2023-12-15 15:05:10.622] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:04 [2023-12-15 15:05:10.622] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:04 [2023-12-15 15:05:10.622]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:04 [2023-12-15 15:05:10.622]
00:00:04 [2023-12-15 15:05:10.622] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:04 [2023-12-15 15:05:10.622]
00:00:04 [2023-12-15 15:05:10.622] eu.mine.zergpool.com (ethashb3)    PING: 89ms    DIFFICULTY: 0.250    EPOCH: 45
00:00:04 [2023-12-15 15:05:10.622]
00:00:04 [2023-12-15 15:05:10.622] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:04 [2023-12-15 15:05:10.622] GPU0    5700XT   OpenCL  -1      0       0/0     46      2040    1840    0
00:00:04 [2023-12-15 15:05:10.622]                                                                          0
00:00:04 [2023-12-15 15:05:10.622] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:04 [2023-12-15 15:05:10.622] GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:04 [2023-12-15 15:05:10.622]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:04 [2023-12-15 15:05:10.622]
00:00:04 [2023-12-15 15:05:10.622] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:04 [2023-12-15 15:05:10.622]
```

**Comment by sp-hash** [comment]
The problem is that the verthash intensity is too high as default. Try to add --dual-intensity 16

**Comment by UselessGuru** [comment]
> The problem is that the verthash intensity is too high as default. Try to add --dual-intensity 16

I expect the miner to work with its default settings.

**Comment by UselessGuru** [comment]
Still the same with 2.16

**Comment by sp-hash** [comment]
try --dual-xintensity 1

**Comment by UselessGuru** [comment]
Still no luck:

`TBMiner.exe --algo etc+ethb3 --hostname etchash.auto.nicehash.com --wallet ************************.*************** --port 9200 --server-passwd x --ethb3-hostname ethashb3.eu.mine.zergpool.com --ethb3-wallet *************************** --ethb3-passwd c=BTC,mc=HYP,ID=*********** --ethb3-port 9996 --api --api-version 1.4 --api-port 4001 --cl-devices [0] --dual-xintensity 1`

```
Cuda runtime API version: 12.0
 ______ _____ _______
|_    _| __  |       |
  |  | | __ -| |  |  |
  |_ | |____ |_|__|  |

Team Black Miner v2.16


GPU0 gfx1010:xnack- 8GB added to miner as OpenCL device
Trying ethashb3.eu.mine.zergpool.com:9996 (ethashb3)
Api server started at 127.0.0.1:4001
Trying etchash.auto.nicehash.com:9200 (etchash)
GPU0 Started CL work thread
GPU0 Started CL work thread
etchash.auto.nicehash.com: Pool responded successfully to subscribe
etchash.auto.nicehash.com: Pool responded successfully to extranonce subscribe
ethashb3.eu.mine.zergpool.com: Pool responded successfully to subscribe
etchash.auto.nicehash.com: Pool responded successfully to authorize
etchash.auto.nicehash.com: Difficulty is set to 0.250000
GPU0 Starting generating etchash dag (Epoch: 315)
ethashb3.eu.mine.zergpool.com: Pool responded successfully to extranonce subscribe
ethashb3.eu.mine.zergpool.com: Extranonce is set to d715000000000000
ethashb3.eu.mine.zergpool.com: Pool responded successfully to authorize
ethashb3.eu.mine.zergpool.com: Difficulty is set to 0.249996
GPU0 Starting generating ethashb3 dag (Epoch: 46)
00:00:01 etchash.auto.nicehash.com (etchash) PING: 31ms DIFFICULTY: 0.250 EPOCH: 315

ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
                                                                         0
ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
                 0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%

SHARES PER MINUTE: 0.00                         POOL HASHRATE: 0.00 H/s

00:00:01 ethashb3.eu.mine.zergpool.com (ethashb3) PING: 95ms DIFFICULTY: 0.250 EPOCH: 46

ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
GPU0    5700XT   OpenCL  -1      1       0/0     46      2040    1840    0
                                                                         0
ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
                 0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%

SHARES PER MINUTE: 0.00                         POOL HASHRATE: 0.00 H/s

etchash.auto.nicehash.com: Extranonce is set to 8000600000000000
etchash.auto.nicehash.com: Difficulty is set to 0.250000
00:00:02 etchash.auto.nicehash.com (etchash) PING: 31ms DIFFICULTY: 0.250 EPOCH: 315

ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
GPU0    5700XT   OpenCL  0       0       0/0     46      2040    1840    0
                                                                         0
ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
                 0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%

SHARES PER MINUTE: 0.00                         POOL HASHRATE: 0.00 H/s

00:00:02 ethashb3.eu.mine.zergpool.com (ethashb3) PING: 96ms DIFFICULTY: 0.250 EPOCH: 46

ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
GPU0    5700XT   OpenCL  -1      1       0/0     46      2040    1840    0
                                                                         0
ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
GPU0    5700XT   0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
                 0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%

SHARES PER MINUTE: 0.00                         POOL HASHRATE: 0.00 H/s

etchash.auto.nicehash.com: Difficulty is set to 0.125000
```

**Comment by UselessGuru** [comment]
Fixed in 2.17
